SDK For Azure.AI.Language.Text

quentinRobinson opened the pull request:
# Contributing to the Azure SDK
New SDK for Azyre AI Language TextAnalyics

quentinRobinson committed:
Add TextAnalytics SDK

quentinRobinson committed:
Merge branch 'main' of https://github.com/quentinRobinson/azure-sdk-for-net into qrobinson/text-analytics-sdk

quentinRobinson committed:
General SDK updates

quentinRobinson committed:
add some live client tests

quentinRobinson committed:
feat: Initial release

quentinRobinson committed:
Add assets.json file for Azure.AI.Language.Text

quentinRobinson committed:
Merge branch 'Azure:main' into qrobinson/text-analytics-sdk

quentinRobinson committed:
chore: Update Azure Cognitive Language tests configuration

quentinRobinson committed:
chore: Rename TextClient.cs to TextAnalysisClient.cs and add headder

quentinRobinson commented:
/azp run net - cognitivelanguage - ci

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.AI.Language.Text](https://apiview.dev/Assemblies/Review/76332000a1a44557b19f7a7eacc39624?revisionId=26f51ac4754f403ea5c40cf205360fb6)

quentinRobinson committed:
chore: Update Azure.AI.Language.Text client library for .NET

quentinRobinson committed:
update links to point to main

quentinRobinson committed:
chore: Update readme

quentinRobinson committed:
update tsp

quentinRobinson committed:
fix snippits

PranshuBansalDev commented on sdk/cognitivelanguage/Azure.AI.Language.Text/samples/Sample10_AnalyzeTextOperationAsync_CustomMultiLabelClassificationOperationAction.md:
```diff
@@ -0,0 +1,93 @@
+{
+    new CustomMultiLabelClassificationOperationAction
+    {
+        Name = "CMCOperationActionSample",
```
Might be good to add a comment here around this being a "string for humans" and that it doesn't need to be carefully selected

PranshuBansalDev commented on sdk/cognitivelanguage/Azure.AI.Language.Text/samples/Sample10_AnalyzeTextOperationAsync_CustomMultiLabelClassificationOperationAction.md:
```diff
@@ -0,0 +1,93 @@
+        foreach (ClassificationDocumentResultWithDetectedLanguage customClassificationDocument in customClassificationResult.Results.Documents)
+        {
+            Console.WriteLine($"Result for document with Id = \"{customClassificationDocument.Id}\":");
+            Console.WriteLine($"  Recognized {customClassificationDocument.Class.Count} classifications:");
```
nit - `\t` instead of spaces?

PranshuBansalDev commented on sdk/cognitivelanguage/Azure.AI.Language.Text/samples/Sample12_AnalyzeTextOperationAsync_AbstractiveSummarizationOperationAction.md:
```diff
@@ -0,0 +1,123 @@
+
+## Summarize one or more text documents
+
+To summarize one or more text documents using extractive summarization, call `AnalyzeTextOperationAsync` on the `TextAnalysisClient` by passing the documents as `MultiLanguageTextInput` parameter and a `AnalyzeTextOperationAction` with a `AbstractiveSummarizationOperationAction` action. This returns a `Response<AnalyzeTextOperationState>` which you can extract the `AbstractiveSummarizationOperati ...
```
Do we need docs to explain the diff between abstractive and extractive summarization?

quentinRobinson replied:
First line gives basic explanation and links to the documentation with examples.

PranshuBansalDev commented on sdk/cognitivelanguage/Azure.AI.Language.Text/samples/Sample13_AnalyzeTextOperationAsync_MultipleActions.md:
````diff
@@ -0,0 +1,143 @@
+
+```C# Snippet:CreateTextClient
+Uri endpoint = new Uri("<your endpoint>");
+AzureKeyCredential credential = new("your apikey");
````
Do we want to think about adding docs on how to call these endpoints with:

1keyv2
managed identity

etc (basically - all the options that are non-key based since we are recommending no more keys)

PranshuBansalDev replied:
Nvm on this one - Bidisha pointed me at this - https://github.com/Azure/azure-sdk-for-net/tree/07243563fcdf1aa9b19f88e6b2952fb111d232ba/sdk/cognitivelanguage/Azure.AI.Language.Conversations#create-a-client-using-azure-active-directory-authentication

quentinRobinson committed:
fix readme

PranshuBansalDev reviewed (commented):
Skimmed through looks good overall!

quentinRobinson committed:
add test for client creation

quentinRobinson committed:
fix sentiment samples

quentinRobinson committed:
fix samples

quentinRobinson commented on sdk/cognitivelanguage/Azure.AI.Language.Text/samples/Sample10_AnalyzeTextOperationAsync_CustomMultiLabelClassificationOperationAction.md:
```diff
@@ -0,0 +1,93 @@
+    {
+        new MultiLanguageInput("A", documentA)
+        {
+            Language = "en"
```
We define this field as a string in the tsp, to change to an enum would be denied for being a breaking change for customers.

quentinRobinson committed:
fix samples

quentinRobinson committed:
fix client test

quentinRobinson committed:
Merge branch 'Azure:main' into qrobinson/text-analytics-sdk

quentinRobinson committed:
update tests

quentinRobinson committed:
add words to cspell ingore list

quentinRobinson committed:
add another word to cspell

quentinRobinson committed:
run codechecks

aurghob commented on sdk/cognitivelanguage/Azure.AI.Language.Text/README.md:
```diff
@@ -0,0 +1,152 @@
+<!-- CLIENT COMMON BAR -->
+[Client options](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#configuring-service-clients-using-clientoptions) |
+[Accessing the response](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#accessing-http-response-details-using-responset) |
+[Long-running operations](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#consuming-long-running-operations-using-operationt) |
```
Do we want to rename Long Running operation to something else? I believe we renamed them to Operation ? Should we be consistent with naming even during presentation and documentation or is it ok to call Long Running Operation here?

quentinRobinson replied:
No, this is referencing the concept of a long running operation, which is provided by azure.

aurghob commented on sdk/cognitivelanguage/Azure.AI.Language.Text/samples/Sample1_AnalyzeText_LanguageDetection.md:
````diff
@@ -0,0 +1,136 @@
+}
+```
+
+See the [README] of the Text Analytics client library for more information, including useful links and instructions.
````
Is this a missed Link ?

aurghob commented on sdk/cognitivelanguage/Azure.AI.Language.Text/samples/Sample2_AnalyzeTextAsync_Sentiment.md:
````diff
@@ -0,0 +1,192 @@
+}
+```
+
+See the [README] of the Text Analytics client library for more information, including useful links and instructions.
````
Missed link again? Seems similar in other samples as well. Maybe I misunderstand but assuming these should have links

aurghob replied:
I'm sorry I forgot this - in md : will [README] get autolinked?

quentinRobinson replied:
Link is referenced at the bottom of the file

aurghob commented on sdk/cognitivelanguage/Azure.AI.Language.Text/samples/Sample5_AnalyzeTextAsync_RecognizePiiEntities.md:
```diff
@@ -0,0 +1,87 @@
+foreach (PiiResultWithDetectedLanguage piiResult in piiTaskResult.Results.Documents)
+{
+    Console.WriteLine($"Result for document with Id = \"{piiResult.Id}\":");
+
```
Do we also want to highlight redactedText in the document here?

quentinRobinson replied:
Added

aurghob commented on sdk/cognitivelanguage/Azure.AI.Language.Text/samples/Sample5_AnalyzeText_RecognizePiiEntities.md:
```diff
@@ -0,0 +1,87 @@
+foreach (PiiResultWithDetectedLanguage piiResult in piiTaskResult.Results.Documents)
+{
+    Console.WriteLine($"Result for document with Id = \"{piiResult.Id}\":");
+
```
Similar to above : perhaps we want to highlight redactedText?

aurghob commented on sdk/cognitivelanguage/Azure.AI.Language.Text/samples/Sample7_AnalyzeTextOperation_HealthcareOperationAction.md:
```diff
@@ -0,0 +1,153 @@
+# Analyze healthcare entities
```
How is this sample different from the analyzeTextOperationAsync for healthcare operation?

quentinRobinson replied:
It is calling the non async function..this sample is mirrored from the old sdk

aurghob replied:
Do we really need this ? Since healthcare doesn't have anything other than Async

quentinRobinson replied:
it is C# notion of async not LRO async

aurghob commented on sdk/cognitivelanguage/Azure.AI.Language.Text/tests/Samples/Sample5_AnalyzeText_RecognizePii.cs:
```diff
@@ -0,0 +1,88 @@
+
+            foreach (PiiResultWithDetectedLanguage piiResult in piiTaskResult.Results.Documents)
+            {
+                Console.WriteLine($"Result for document with Id = \"{piiResult.Id}\":");
```
Add redacted text as well ?

quentinRobinson committed:
address PR comments

quentinRobinson committed:
fix main readme

quentinRobinson commented on sdk/cognitivelanguage/Azure.AI.Language.Text/CHANGELOG.md:
```diff
@@ -0,0 +1,13 @@
+## 1.0.0-beta.1 (Unreleased)
+
+- Initial release
+
```
Just going to add API versions supported for this SDK

quentinRobinson committed:
fix readme

quentinRobinson committed:
remove model namespace

quentinRobinson committed:
chore: Update Azure.AI.Language.Text.Tests namespace in Sample4_AnalyzeText_RecognizeEntities.cs

quentinRobinson committed:
remove broken links to fix ci

quentinRobinson committed:
remove local from link

quentinRobinson committed:
Merge branch 'Azure:main' into qrobinson/text-analytics-sdk

quentinRobinson committed:
fix build errors

quentinRobinson committed:
Merge branch 'Azure:main' into qrobinson/text-analytics-sdk

bidisha-c commented on sdk/cognitivelanguage/Azure.AI.Language.Text/README.md:
```diff
@@ -0,0 +1,268 @@
+
+### Install the package
+
+First, provide instruction for obtaining and installing the package or library. This section might include only a single line of code, like `dotnet add package package-name`, but should enable a developer to successfully install the package from NuGet, npm, or even cloning a GitHub repository.
```
Update. Check the conversations ot existing TA SDK README

bidisha-c replied:
We need a migration guide. For reference you can check the QnAMaker SDK

bidisha-c commented on sdk/cognitivelanguage/Azure.AI.Language.Text/README.md:
````diff
@@ -0,0 +1,268 @@
+```C# Snippet:TextAnalysisClientt_BadRequest
+try
+{
+    string documentA =
````
Use "text" instead of document as we have the NDS

quentinRobinson committed:
Update repository URL in tsp-location.yaml to azure main

bidisha-c commented on sdk/cognitivelanguage/Azure.AI.Language.Text/src/Generated/AILanguageTextClientBuilderExtensions.cs:
```diff
@@ -0,0 +1,46 @@
+namespace Microsoft.Extensions.Azure
+{
+    /// <summary> Extension methods to add <see cref="TextAnalysisClient"/> to client builder. </summary>
+    public static partial class AILanguageTextClientBuilderExtensions
```
Let's add a customization to rename this to "TextAnalysisClientBuilderExtensions". We did this for Conversations.

bidisha-c commented on sdk/cognitivelanguage/Azure.AI.Language.Text/src/Generated/AILanguageTextModelFactory.cs:
```diff
@@ -0,0 +1,1700 @@
+
+// <auto-generated/>
+
+#nullable disable
```
same for this one

bidisha-c reviewed (commented):
Additionally can you check the configuration in the Conversations SDK. Most of the generated files are inside a directory model (still keeping the namespace of the root level). I think it makes it easier to find the client class within all model classes.

bidisha-c commented on sdk/cognitivelanguage/Azure.AI.Language.Text/Azure.AI.Language.Text.sln:
```diff
@@ -0,0 +1,37 @@
+		Debug|Any CPU = Debug|Any CPU
+		Release|Any CPU = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
```
Either is OK but I am thinking should we have a separate solution for Text or just add to the existing Language solution?

quentinRobinson replied:
It was created during the setup of the project. Its in both and its nice to load the single SDK from the solution

quentinRobinson committed:
adress pr comments

quentinRobinson commented:
> Additionally can you check the configuration in the Conversations SDK. Most of the generated files are inside a directory model (still keeping the namespace of the root level). I think it makes it easier to find the client class within all model classes.

It was stated in a review comment that the models should be in the same directory as the client.

quentinRobinson committed:
Rename `Documents` to `MultiLanguageInputs`/`LanguageInputs`

Renamed the `Documents` property to `MultiLanguageInputs` or `LanguageInputs` across various files and classes to enhance clarity and consistency in naming conventions. This change affects multiple classes, including `MultiLanguageTextInput` and `LanguageDetectionTextInput`, as well as their serialization methods. Updated files include `README.md`, several sample and test files, and `tsp-location.yaml` to reflect the new naming conventions and repository details.

quentinRobinson committed:
code check fixes

quentinRobinson committed:
rename document to text

quentinRobinson committed:
Rename repository URL in tsp-location.yaml to Azure main
